Basic functionality

maydek1 opened the pull request.

maydek1 committed:
feat(microservice): add basic functionality

- Implemented core CRUD operations
- Added initial controllers and services

psidarovich commented on src/main/java/com/software/modsen/rideservice/dto/response/RideResponseSet.java:
```diff
@@ -0,0 +1,16 @@
+@AllArgsConstructor
+@Getter
+@Setter
+public class RideResponseSet {
+    Set<RideResponse> rideResponseSet;
+}
```
It's better to rename the field. For example, `items`

maydek1 replied:
Renamed

psidarovich commented on src/main/java/com/software/modsen/rideservice/repositories/RideRepository.java:
```diff
@@ -0,0 +1,16 @@
+@Repository
+public interface RideRepository extends JpaRepository<Ride, Long> {
+    Set<Ride> findAllByStatus(RideStatus status);
+    Set<Ride> findAllByDriverId(Long driver_id);
```
Please, use camel case naming instead of snake case

maydek1 replied:
Renamed

psidarovich commented on src/main/java/com/software/modsen/rideservice/service/RideService.java:
```diff
@@ -0,0 +1,22 @@
+import com.software.modsen.rideservice.dto.response.RideResponseSet;
+import com.software.modsen.rideservice.model.RideStatus;
+
+public interface RideService {
```
Creating interfaces for only one implementation is a bad practice unless you have hierarchy or several implementations

maydek1 replied:
Removed the interface

psidarovich commented on src/main/java/com/software/modsen/rideservice/service/ServiceImpl/RideServiceImpl.java:
```diff
@@ -0,0 +1,100 @@
+public class RideServiceImpl implements RideService {
+
+    private final RideRepository rideRepository;
+    private final RideMapper rideMapper;
```
Minor: usually mapping logic goes on controller layer

maydek1 replied:
Moved the mapping logic to the controller layer

maydek1 committed:
refactor: rename variables, move mapping to controller, remove service interface

- Renamed variables for better clarity
- Moved DTO mapping logic from service to controller layer
- Removed unnecessary service interface to simplify the code
